client/components/Button.tsx

```diff
@@ -2,7 +2,7 @@ import { faSpinner } from '@fortawesome/free-solid-svg-icons';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 
 export interface ButtonProps {
-  onClick?: () => void;
+  onClick?: (prop?: any) => void;
```

Comment:
this is a bit of a smell, I don't think it's necessary based on any usage I see so far

I don't typically see cases where we use the parameters from a button's onClick event, rather than `any` this would be something vaguely resembling `OnClickEvent<HTMLButton>`